### PR TITLE
en: add a document about scenarios

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What is changed, added or deleted? (Required)

adding usage scenario

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-dm/pull/88

### CODE REVIEW

File: en/scenarios.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+### Migrating some business data in MySQL/MariaDB
+
+If there are multiple business data in the original MySQL/MariaDB, but only part of the data needs to be migrated to TiDB, you can configure the migration task according to [Use TiDB as a secondary library of MySQL/MariaDB](#tidb-as-mysql/mariadb's-secondary-library), and then configure `black-white-list` as needed.
+
+If you want to migrate upstream data to a schema or table with a different name in downstream, you can additionally configure `route-rules`.
+
+For some archiving scenarios, some data may be periodically cleaned up through `TRUNCATE TABLE`/`DROP TABLE` or other methods in the upstream, but it is expected that all data will be kept in the downstream TiDB, then additional configuration `filter-rules` can be used to filter out these operations.
+
+For such scenarios, please refer to [Data Migration Simple Use Scenario](usage-scenario-simple-replication.md).
```
Comment (toutdesuite) [inline en/scenarios.md:31]:
```suggestion
For more information, refer to [Data Migration Simple Usage Scenario](usage-scenario-simple-replication.md).
```

File: en/scenarios.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Shard-merge scenario
```
Comment (toutdesuite) [inline en/scenarios.md:33]:
```suggestion
## Shard Merge scenario
```

File: en/scenarios.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+If there are multiple sharding tables in the upstream MySQL/MariaDB, it is usually expected to merge them into only one table when migrating to TiDB. The schema name or the table name in the upstream data can be renamed by configuring `route-rules`, and then these tables will be merged into the same downstream schema or table. For details, please refer to [Data Migration Shard Merge Scenario](usage-scenario-shard-merge.md) and [Best Practices of Data Migration in the Shard Merge Scenario](shard-merge-best-practices.md).
```
Comment (toutdesuite) [inline en/scenarios.md:35]:
```suggestion
If there are multiple sharded tables in multiple sharded schemas in the upstream MySQL/MariaDB, to merge them into one table or schema when migrating to TiDB, you can rename the table name or the schema name in the upstream database by configuring `route-rules`, and then these tables can be merged into the same downstream schema or table. For details, refer to [Data Migration Shard Merge Scenario](usage-scenario-shard-merge.md) and [Best Practices of Data Migration in the Shard Merge Scenario](shard-merge-best-practices.md).
```

File: en/scenarios.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+DM supports the migration of DDL specifically. For details, please refer to [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md).
```
Comment (toutdesuite) [inline en/scenarios.md:37]:
```suggestion
Specifically, DM supports the migration of DDL. For details, refer to [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md).
```

File: en/scenarios.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+If you only need to migrate part of the business data or filter some specific operations, you can directly refer to [Migrating some business data in MySQL/MariaDB](#migrating-some-business-data-in-mysql/mariadb) in part `Non-merging sharding table scenario`.
```
Comment (toutdesuite) [inline en/scenarios.md:39]:
```suggestion
If you only need to migrate some application data or filter out some operations, refer to [Migrating some business data in MySQL/MariaDB](#migrating-some-business-data-in-mysql/mariadb) in part `Non-merging sharding table scenario`.
```

File: en/scenarios.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+## Online DDL scenario
+
+In MySQL's ecosystem, tools such as pt-osc or gh-ost are often used to perform online DDL operations. DM provides special support for such scenarios, which can be enabled by configuring the `online-ddl-scheme` parameter. For details, please refer to [DM online-ddl-scheme](online-ddl-scheme.md).
```
Comment (toutdesuite) [inline en/scenarios.md:43]:
```suggestion
In MySQL ecosystem, tools such as pt-osc or gh-ost are often used to perform online DDL operations. DM also supports such scenarios. You can enable pt-osc and gh-ost by configuring the `online-ddl-scheme` parameter. For details, refer to [DM Online DDL Scheme](online-ddl-scheme.md).
```

File: en/scenarios.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+## Change upstream MySQL instance of DM connection
+
+When using MySQL/MariaDB, a primary-secondary cluster is often built to improve read performance and ensure data security. If the upstream MySQL/MariaDB instance connected by DM-worker is unavailable for some reason when using DM to migration data, you need to connect DM-worker to another instance in the upstream primary-secondary cluster. For such scenarios, DM provides nice support, but there are still some limitations.
```
Comment (toutdesuite) [inline en/scenarios.md:47]:
```suggestion
When you use MySQL/MariaDB, a primary-secondary cluster can be built to improve the read performance and to ensure data safety. If the upstream MySQL/MariaDB instance connected by DM-worker is unavailable for some reason when you use DM to migrate data, you need to connect DM-worker to another instance in the upstream primary-secondary cluster. For such scenarios, DM provides good support, but there are some limitations.
```

File: en/scenarios.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+DM supports the migration of DDL specifically. For details, please refer to [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md).
+
+If you only need to migrate part of the business data or filter some specific operations, you can directly refer to [Migrating some business data in MySQL/MariaDB](#migrating-some-business-data-in-mysql/mariadb) in part `Non-merging sharding table scenario`.
+
+## Online DDL scenario
+
+In MySQL's ecosystem, tools such as pt-osc or gh-ost are often used to perform online DDL operations. DM provides special support for such scenarios, which can be enabled by configuring the `online-ddl-scheme` parameter. For details, please refer to [DM online-ddl-scheme](online-ddl-scheme.md).
+
+## Change upstream MySQL instance of DM connection
```
Comment (toutdesuite) [inline en/scenarios.md:45]:
```suggestion
## Switch the DM-worker connection to another instance
```

File: en/scenarios.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+category: reference
+---
+
+# Usage scenarios
+
+This document introduces the main usage scenarios supported by TiDB Data Migration (DM) and related usage suggestions.
+
+## Non-shard-merge scenario
+
+### TiDB as MySQL/MariaDB's secondary library
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply en/scenarios.md:13]: most of the documents use `secondary library`, should we unified the word?

File: en/scenarios.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+# Usage Scenarios
+
+This document introduces the main usage scenarios of TiDB Data Migration (DM) and the usage suggestions.
+
+## Non-shard-merge scenario
+
+### Use TiDB as MySQL/MariaDB's secondary database
+
+If you want to use TiDB as the upstream MySQL/MariaDB's secondary library, that is, import the full data of the upstream instance to TiDB, and replicate the incremental data to TiDB in real-time once data changes are made, you can configure the data migration task according to the following rules:
```
Comment (csuzhangxc) [inline en/scenarios.md:15]: How about using always using `secondary database`?

File: en/scenarios.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+- Specify `task-mode` as `all`.
+- Configure `target-database` based on the connection information of the downstream TiDB.
+- Configure `source-id` in `mysql-instances` based on the ID information of the upstream MySQL/MariaDB instances.
+- Use the default setting of concurrency control parameters or configure `mydumper-thread`, `loader-thread`, and `syncer-thread` as needed.
+- No need to configure `route-rules`, `filter-rules` and `black-white-list`.
```
Comment (csuzhangxc) [inline en/scenarios.md:21]: How about updating `black-white-list` in this PR?

File: en/scenarios.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+### Migrate some application data from MySQL/MariaDB
+
+If there are multiple application data in MySQL/MariaDB, but only part of the data needs to be migrated to TiDB, you can configure the migration task by referring to [Use TiDB as a secondary library of MySQL/MariaDB](#use-tidb-as-mysqlmariadbs-secondary-database). Then configure `black-white-list` as needed.
+
+To migrate upstream data to a schema or table with a different name in the downstream database, you can configure `route-rules`.
+
+For some archiving scenarios, some data may be periodically cleaned up by executing the `TRUNCATE TABLE`/`DROP TABLE` command  in the upstream or through other means. To make sure all data are retained in the downstream TiDB, you can disable these data clearing operations by configuring `filter-rules`.
```
Comment (csuzhangxc) [inline en/scenarios.md:29]:
```suggestion
For some archiving scenarios, some data may be periodically cleaned up by executing the `TRUNCATE TABLE`/`DROP TABLE` command in the upstream or through other means. To make sure all data are retained in the downstream TiDB, you can disable these data clearing operations by configuring `filter-rules`.
```

File: en/scenarios.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+Specifically, DM supports the migration of DDL. For details, refer to [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md).
+
+If you only need to migrate some application data or filter out some operations, refer to [Migrate some applicationb data from MySQL/MariaDB](#migrate-some-application-data-from-mysqlmariadb) in part `Non-merging sharding table scenario`.
+
+## Online DDL scenario
+
+In MySQL ecosystem, tools such as pt-osc or gh-ost are often used to perform online DDL operations. DM also supports such scenarios. You can enable pt-osc and gh-ost by configuring the `online-ddl-scheme` parameter. For details, refer to [DM Online DDL Scheme](online-ddl-scheme.md).
```
Comment (csuzhangxc) [inline en/scenarios.md:43]:
```suggestion
In MySQL ecosystem, tools such as pt-osc or gh-ost are often used to perform online DDL operations. DM also supports such scenarios. You can enable support for pt-osc or gh-ost by configuring the `online-ddl-scheme` parameter. For details, refer to [DM Online DDL Scheme](online-ddl-scheme.md).
```

File: en/scenarios.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+## Switch the DM-worker connection to another instance
+
+When you use MySQL/MariaDB, a primary-secondary cluster can be built to improve the read performance and to ensure data safety. If the upstream MySQL/MariaDB instance connected by DM-worker is unavailable for some reason when you use DM to migrate data, you need to switch the DM-worker connection to another MySQL/MariaDB instance in the upstream primary-secondary cluster. For such scenarios, DM provides good support, but there are some limitations.
```
Comment (csuzhangxc) [inline en/scenarios.md:47]: `usage-scenario-master-slave-switch.md` has not been translated yet?

Comment (WangXiangUSTC) [reply en/scenarios.md:47]: yes

**Comment by WangXiangUSTC** [comment]
@csuzhangxc all addressed in https://github.com/pingcap/docs-dm/pull/134/commits/8e63cb71f4726aca35caaf1cf27c045f65e3d992, PTAL again

**Comment by WangXiangUSTC** [comment]
> LGTM. but seem I forgot to add the doc to TOC in #88.

already add in TOC with Chineses version, the English version TOC is not reconstructed, maybe we can update it after most of the English document is ready.

**Comment by toutdesuite** [comment]
> > LGTM. but seem I forgot to add the doc to TOC in #88.
> 
> already add in TOC with Chineses version, the English version TOC is not reconstructed, maybe we can update it after most of the English document is ready.

TOC updated in https://github.com/pingcap/docs-dm/pull/133

File: en/scenarios.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
+---
+title: Usage Scenarios
+summary: Understand the main usage scenarios of TiDB Data Migration.
+category: reference
```
Comment (toutdesuite) [inline en/scenarios.md:4]:
```suggestion
```

File: en/scenarios.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+To migrate upstream data to a schema or table with a different name in the downstream database, you can configure `route-rules`.
+
+For some archiving scenarios, some data may be periodically cleaned up by executing the `TRUNCATE TABLE`/`DROP TABLE` command in the upstream or through other means. To make sure all data are retained in the downstream TiDB, you can disable these data clearing operations by configuring `filter-rules`.
+
+For more information, refer to [Data Migration Simple Usage Scenario](usage-scenario-simple-replication.md).
+
+## Shard Merge scenario
+
+If there are multiple sharded tables in multiple sharded schemas in the upstream MySQL/MariaDB, to merge them into one table or schema when migrating to TiDB, you can rename the table name or the schema name in the upstream database by configuring `route-rules`, and then these tables can be merged into the same downstream schema or table. For details, refer to [Data Migration Shard Merge Scenario](usage-scenario-shard-merge.md) and [Best Practices of Data Migration in the Shard Merge Scenario](shard-merge-best-practices.md).
```
Comment (toutdesuite) [inline en/scenarios.md:34]:
```suggestion
If there are multiple sharded tables in multiple sharded schemas in the upstream MySQL/MariaDB, to merge them into one table or schema when migrating to TiDB, you can rename the table or the schema in the upstream database by configuring `route-rules`, and then these tables can be merged into the same downstream schema or table. For details, refer to [Data Migration Shard Merge Scenario](usage-scenario-shard-merge.md) and [Best Practices of Data Migration in the Shard Merge Scenario](shard-merge-best-practices.md).
```

File: en/scenarios.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+Specifically, DM supports the migration of DDL. For details, refer to [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md).
+
+If you only need to migrate some application data or filter out some operations, refer to [Migrate some applicationb data from MySQL/MariaDB](#migrate-some-application-data-from-mysqlmariadb) in part `Non-merging sharding table scenario`.
```
Comment (toutdesuite) [inline en/scenarios.md:38]:
```suggestion
If you only need to migrate some application data or filter out some operations, refer to [Migrate some applicationb data from MySQL/MariaDB](#migrate-some-application-data-from-mysqlmariadb).
```

**Pull request merged**